base/src/commonMain/kotlin/utils/Decimal.kt

```diff
@@ -76,26 +136,54 @@ expect fun Decimal.minus(value: Decimal, scale: Int): Decimal
  * @param roundingMode The [RoundingMode] to apply when scaling.
  * @return the [Decimal] that is the subtraction of the two provided decimals.
  */
-expect fun Decimal.minus(
+fun Decimal.minus(
     value: Decimal,
     scale: Int,
     roundingMode: RoundingMode = RoundingMode.RoundHalfEven
-): Decimal
+): Decimal = if (this is Decimal.Finite && value is Decimal.Finite) {
+    Decimal.Finite(finiteDecimal.minus(value.finiteDecimal, scale, roundingMode))
+} else {
+    (this.toDouble() - value.toDouble()).toDecimal()
+}
+
+internal expect operator fun FiniteDecimal.minus(value: FiniteDecimal): FiniteDecimal
+internal expect fun FiniteDecimal.minus(value: FiniteDecimal, scale: Int): FiniteDecimal
+internal expect fun FiniteDecimal.minus(
+    value: FiniteDecimal,
+    scale: Int,
+    roundingMode: RoundingMode = RoundingMode.RoundHalfEven
+): FiniteDecimal
 
 /**
  * Divides two [Decimal].
  * @param value the [Decimal] to divide
  * @return the [Decimal] that is the division of the two provided decimals.
  */
-expect operator fun Decimal.div(value: Decimal): Decimal
+operator fun Decimal.div(value: Decimal): Decimal = if (this is Decimal.Finite && value is Decimal.Finite) {
+    if (value.toDouble() == 0.0) {
+        (toDouble() / value.toDouble()).toDecimal()
```

Comment:
Here you explicitly divide by 0

Reply:
Being consistent with Kotlin.Double
https://pl.kotl.in/FQsIIQzI4

Reply:
For Kotlin.Double I do see warnings in such cases, is it possible to do so for our Decimal?